spooq/transformer/base_cleaner.py

```diff
@@ -37,8 +37,7 @@ def _add_temporary_columns(input_df, column_names, temporary_column_names):
     def _log_cleansed_values(self, input_df, column_names, temporary_column_names):
         def _only_keep_cleansed_values(col_name, temporary_col_name):
             return F.when(F.col(temporary_col_name) == F.col(col_name), F.lit(None)).otherwise(
-                F.col(temporary_col_name)
-            )
+                F.when(F.col(temporary_col_name).isNull(), F.lit("null")).otherwise(F.col(temporary_col_name)))
```

Comment:
This will also create an entry in the log column if you use the Threshold Cleaner and the input value is NULL.
Please create a test for this (just create a new input dataframe, even I find the setup of the Threshold Cleaner confusing :laughing: )

Reply:
To avoid casting everything to a string, you could try to get the original data type and cast the original value. Maybe something like this (Not sure if it will work ;-) ):
```
def _only_keep_cleansed_values(col_name, temporary_col_name, original_data_type):
    return F.when(F.col(temporary_col_name) == F.col(col_name), F.lit(None)).otherwise(
        F.when(F.col(temporary_col_name).isNull(), F.lit("null")).otherwise(F.col(temporary_col_name).cast(original_data_type())))

for column_name, temporary_column_name in zip(column_names, temporary_column_names):
    # Only keep cleansed values in temporary columns
    original_data_type = input_df.schema[column_name].dataType
    input_df = input_df.withColumn(
        temporary_column_name, _only_keep_cleansed_values(column_name, temporary_column_name, original_data_type)
    )
```

Reply:
New tests are called: test_cleansing_log_when_input_value_is_null. I did one for struct and one for map.